code-name-taran/src/com/spaghettisoft/component/dontgetangry/DontGetAngry.java

```diff
@@ -0,0 +1,253 @@
+package com.spaghettisoft.component.dontgetangry;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import com.spaghettisoft.component.game.AbstractGame;
+import com.spaghettisoft.globals.StaticObjects;
+
+public class DontGetAngry extends AbstractGame{
+
+	private State currentState = State.STARTING;
+	static List<Player> players;
+	static Dice dice;
+	Player player;
+	private boolean ended;
+	
+	
+	@Override
+	public void show() {
+		initialize();
+		super.show();
+	}
+	
+	private void initialize() {
+		ended = false;
+	}
+
+	public DontGetAngry() {
+		dice = new Dice(6);
+		ended = false;
+	}
+
+	@Override
+	protected void drawGame() {
+		if(currentState == State.STARTING){
+			setPlayerPositions();
+			currentState = State.PLAYING;
+		}
+		Board.print();
+	}
+	
+	@Override
+	protected boolean isEnded() {
+		return ended;
+	}
+	
+	@Override
+	protected void processGame() {
+		
+		for (int k = 0; k < players.size(); k++) {
+			player = players.get(k);
+			int squares = dice.roll(player);
+			turn(player, squares);
+			if(player.hasWon()){
+				gameEnds();
+				break;
+			}
+		}
+	}
+	
+	@Override
+	protected void printEndGameMessage() {
+		System.out.println("Congratulations "+ player.name+"!!!");
+		System.out.println("You have won the game!");
+	}
+	/*
+	public void play() {
+		
+		setPlayerPositions();
+		drawGame();
+		
+		while (!isEnded()) {
+			processGame();
+			
+		}
+		printEndGameMessage();
+
+	}
+	/*
+	public void start(){
+		Rules rules = new Rules();
+		
+		int choice;
+		do {
+			choice = game.getIntegerOption();
+			switch (choice) {
+			case 1:
+				game.play();
+				break;
+			case 2:
+				rules.show();
+				break;
+			case 0:
+				break;
+			default:
+				System.out.println("Incorrect option.");
+				break;
+			}
+		} while (choice != 0);
+	}
+
+	*/
+	public void updatePlayerSquares(Token token){
+		if(token.hasFinished()){
+			token.player.diagSquaresLeft--;
+			token.player.inGameTokens.remove(token);
+		}
+	}
+	
+	public void checkForCollisionsWithOtherTokens(Token token) {
+		for (int i = 0; i < players.size(); i++) {
+			Player player = players.get(i);
+			if (player != token.player) {
+				for (int j = 0; j < player.inGameTokens.size(); j++) {

```

Comment:
@georgiatanasov147 
Try to avoid nesting so many loops inside each other. It makes the code less readable. You can split them in several methods and call them inside of each other ^_^